pkg/operator/controllers/autosizednodes/autosizednodes_controller_test.go

```diff
@@ -0,0 +1,145 @@
+	// ARO unfortunately relies on implicit import and its side effect for this
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	_ "github.com/Azure/ARO-RP/pkg/util/scheme"
+)
+
+func TestAutosizednodesReconciler(t *testing.T) {
+	aroMeta := metav1.ObjectMeta{
+		Name:      "aro",
+		Namespace: "openshift-azure-operator",
+	}
+
+	emptyConfig := mcv1.KubeletConfig{}
+	config := makeConfig()
+
+	tests := []struct {
+		name       string
+		wantResult ctrl.Result
+		wantErr    error
+		wantGetErr error
+		client     client.Client
+		wantConfig *mcv1.KubeletConfig
+	}{
+		{
+			name:       "is not needed",
+			wantResult: ctrl.Result{},
+			client: fake.NewClientBuilder().WithRuntimeObjects(
+				&arov1alpha1.Cluster{
+					ObjectMeta: aroMeta,
+					Spec: arov1alpha1.ClusterSpec{
+						Features: arov1alpha1.FeaturesSpec{
+							ReconcileAutoSizedNodes: false,
+						},
+					},
+				},
+			).Build(),
+			wantConfig: &emptyConfig,
+			wantErr:    nil,
+			wantGetErr: kerrors.NewNotFound(mcv1.Resource("kubeletconfigs"), "dynamic-node"),
+		},
+		{
+			name:       "is needed and not present already",
+			wantResult: ctrl.Result{},
+			client: fake.NewClientBuilder().WithRuntimeObjects(
+				&arov1alpha1.Cluster{
+					ObjectMeta: aroMeta,
+					Spec: arov1alpha1.ClusterSpec{
+						Features: arov1alpha1.FeaturesSpec{
+							ReconcileAutoSizedNodes: true,
+						},
+					},
+				}).Build(),
+			wantConfig: &config,
+			wantErr:    nil,
+			wantGetErr: nil,
+		},
+		{
+			name:       "is needed and present already",
+			wantResult: ctrl.Result{},
+			wantErr:    nil,
+			client: fake.NewClientBuilder().WithRuntimeObjects(
+				&arov1alpha1.Cluster{
+					ObjectMeta: aroMeta,
+					Spec: arov1alpha1.ClusterSpec{
+						Features: arov1alpha1.FeaturesSpec{
+							ReconcileAutoSizedNodes: true,
+						},
+					},
+				},
+				&config).Build(),
+			wantConfig: &config,
+		},
+		{
+			name:       "is not needed and is present",
+			wantResult: ctrl.Result{},
+			client: fake.NewClientBuilder().WithRuntimeObjects(
+				&arov1alpha1.Cluster{
+					ObjectMeta: aroMeta,
+					Spec: arov1alpha1.ClusterSpec{
+						Features: arov1alpha1.FeaturesSpec{
+							ReconcileAutoSizedNodes: false,
+						},
+					},
+				},
+				&config).Build(),
+			wantConfig: &emptyConfig,
+			wantErr:    nil,
+			wantGetErr: kerrors.NewNotFound(mcv1.Resource("kubeletconfigs"), "dynamic-node"),
+		},
+	}
+
+	cluster := arov1alpha1.Cluster{}
+	cluster.DeepCopyObject()
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx := context.Background()
+			r := Reconciler{
+				Client: test.client,
+				log:    logrus.NewEntry(logrus.StandardLogger()),
+			}
+			result, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: types.NamespacedName{Namespace: "openshift-azure-operator", Name: "aro"}})
+			if err != test.wantErr {
+				t.Errorf("reconcile returned an unexpected error, want: %v, got: %v", test.wantErr, err)
+			}
```

Comment:
1. I see that none of the test cases have `wantErr`. I think we are not testing cases when `Reconcile` returns an error.
1. Nit: this will give enough context (file name, line number and test case name) to debug tests:
    ```suggestion
    			if err != test.wantErr {
    				t.Error(err)
    			}
    ```